aus_council_scrapers/base.py

```diff
@@ -21,132 +21,14 @@
-    `location`: The location of the meeting (e.g. Council Chambers).\n
-    `cleaned_time`: The time of the meeting as a time object.\n
-    `cleaned_date`: The date of the meeting as a date object.\n
-    """
-
-    name: Optional[str]
-    date: str
-    time: Optional[str]
-    webpage_url: str
-    download_url: str
-    location: Optional[str] = None
-
-    # Cached properties
-    _cleaned_time: Optional[datetime.time] = None
-    _cleaned_date: Optional[datetime.date] = None
-
-    @property
-    def cleaned_time(self) -> Optional[datetime.time]:
-        try:
-            if not self.time:
-                return None
-            if not self._cleaned_time:
-                self._cleaned_time = parse_date(self.time, fuzzy=True).time()
-            return self._cleaned_time
-        except Exception as e:
-            return None
-
-    @property
-    def cleaned_date(self) -> datetime.date:
-        if not self.date:
-            raise ValueError("Date is required")
-
-        try:
-            if not self._cleaned_date:
-                self._cleaned_date = parse_date(self.date, fuzzy=True).date()
-        except Exception as e:
-            raise ValueError(f"Could not parse date {self.date}")
-
-        return self._cleaned_date
-
-    @property
-    def cleaned_location(self) -> Optional[str]:
-        if not self.location or self.location.isspace():
-            return None
-
-        cleaned = self.location.replace(r"\w", " ").strip().lower()
-
-        # Remove council chambers string from location
-        council_chamber_regex = re.compile(r"^council\s?chambers?,?", re.IGNORECASE)
-        cleaned = council_chamber_regex.sub("", cleaned)
-
-        if cleaned == "":
-            return None
-
-        return " ".join((word.capitalize() for word in cleaned.split()))
-
-    def check_required_properties(self, state: str) -> None:
-        if not self.name or self.name.isspace():
-            raise ValueError(f"No name found")
-        if not self.download_url or self.download_url.isspace():
-            raise ValueError(f"No download URL found")
-        if not self.webpage_url or self.webpage_url.isspace():
-            raise ValueError(f"No webpage URL found")
-
-        # cleaned date check happens in the property getter
-        _ = self.cleaned_date
-
-        # Check if date is in the past
-        # TODO: Do we want to add this check to make sure we're not scraping meetings that happened in the past?
-        # if self.is_date_in_past(state):
-        #     raise ValueError(f"Meeting date is in the past")
-
-    def add_default_values(self, default_name, default_time, default_location):
-        if not self.name and default_name:
-            self.name = default_name
-        if not self.time and default_time:
-            self.time = default_time
-        if not self.cleaned_location and default_location:
-            self.location = default_location
-
-    def is_date_in_past(self, state: str) -> bool:
```

Comment:
I was being pretentious and called it `aus_council_scrapers.data.NoticeDate.Base ` in